File: daily_problems/2024/07/0716/personal_submission/cf442b_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3")
using namespace std;

signed main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    vector<double> probs(n);
    for (auto &prob: probs) cin >> prob;
    
    sort(probs.begin(), probs.end(), [&](double x, double y) {return x > y;});

    double p0 = 1, p1 = 0, ans = 0;

    for (auto &prob: probs)
        p1 = p1 * (1 - prob) + p0 * prob, p0 *= 1 - prob, ans = max(ans, p1);

    cout << setprecision(15) << ans;

    return 0;
}

File: daily_problems/2024/07/0716/personal_submission/cf442b_yawn_sean.py
# Submission Link: https://codeforces.com/contest/442/submission/270783399
def main():
    n = II()
    probs = LFI()
    probs.sort(reverse=True)

    p0, p1 = 1, 0
    ans = 0

    for p in probs:
        p1 = p1 * (1 - p) + p0 * p
        p0 *= 1 - p
        ans = max(ans, p1)

    print(ans)